cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancement/queries.spec.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import {
+  MiscUtils,
+  TestFixtureHandler,
+} from '@opensearch-dashboards-test/opensearch-dashboards-test-library';
+import { CURRENT_TENANT } from '../../../../../utils/commands';
+
+const miscUtils = new MiscUtils(cy);
+const testFixtureHandler = new TestFixtureHandler(
+  cy,
+  Cypress.env('openSearchUrl')
+);
+
+const indexSet = [
+  'logstash-2015.09.22',
+  'logstash-2015.09.21',
+  'logstash-2015.09.20',
+];
+
+describe('query enhancement queries', { scrollBehavior: false }, () => {
+  before(() => {
+    CURRENT_TENANT.newTenant = 'global';
+    cy.fleshTenantSettings();
+    cy.deleteAllIndices();
+    cy.deleteSavedObjectByType('index-pattern');
+
+    testFixtureHandler.importJSONMapping(
+      'cypress/fixtures/dashboard/opensearch_dashboards/query_enhancement/mappings.json.txt'
+    );
+
+    testFixtureHandler.importJSONDoc(
+      'cypress/fixtures/dashboard/opensearch_dashboards/query_enhancement/data_with_index_pattern.json.txt'
+    );
+
+    // Go to the Discover page
+    miscUtils.visitPage(`app/data-explorer/discover#/`);
+
+    cy.setAdvancedSetting({
+      defaultIndex: 'timestamp-*',
+    });
+
+    // Go to the Discover page
+    miscUtils.visitPage(
+      `app/data-explorer/discover#/?_g=(filters:!(),time:(from:'2018-09-19T13:31:44.000Z',to:'2019-09-24T01:31:44.000Z'))`
+    );
+
+    cy.get(`[class~="datasetSelector__button"]`).click();
+    cy.get(`[data-test-subj="datasetOption-timestamp-*"]`).click();
+
+    cy.waitForLoaderNewHeader();
+    cy.waitForSearch();
+  });
+
+  describe('send queries', () => {
+    it('with DQL', function () {
+      const query = `_id:1`;
+      cy.setSingleLineQueryEditor(query);
+      cy.waitForLoaderNewHeader();
+      cy.waitForSearch();
+      cy.verifyHitCount(1);
+
+      //query should persist across refresh
+      cy.reload();
+      cy.verifyHitCount(1);
+    });
+
+    it('with Lucene', function () {
+      cy.get(`[data-test-subj="queryEditorLanguageSelector"]`).click();
+      cy.get(`[class~="languageSelector__menuItem"]`).eq(1).click({
+        force: true,
+      });
+
+      const query = `_id:1`;
+      cy.setSingleLineQueryEditor(query);
+      cy.waitForLoaderNewHeader();
+      cy.waitForSearch();
+      cy.verifyHitCount(1);
+
+      //query should persist across refresh
+      cy.reload();
+      cy.verifyHitCount(1);
+    });
+
+    it('with PPL', function () {
+      cy.get(`[data-test-subj="queryEditorLanguageSelector"]`).click();
+      cy.get(`[class~="languageSelector__menuItem"]`).eq(2).click({
+        force: true,
+      });
+
+      // default PPL query should be set
+      cy.waitForLoaderNewHeader();
+      cy.waitForSearch();
+      cy.verifyHitCount(4);
```

Comment:
could we update the PPL query in this test and then refresh. 

and i think at minimum we could verify the default query includes `source = ${current dataset}` i wouldn't check the full query since the default query could change over time but i dont think the `source = ${curr dataset}` part will ever change

Reply:
added